[Docs] Site deploy broken since ~July 14: summarizing storage page missing from Swift sidebar

## Problem

Every "Build and Deploy Documentation" run on main has failed since ~2026-07-14:

```
[ERROR] Caught error rendering /swift/storage/built-in/summarizing: TypeError: Cannot convert undefined or null to object
```

Root cause: `docs/src/content/docs/swift/storage/built-in/summarizing.md` was added without a matching sidebar entry in `docs/astro.config.mjs`. The `starlight-sidebar-topics` middleware throws for any page not attached to a topic (its fallback does `Object.entries(options.topics)` with no `topics` option configured).

Consequence: the live docs site is ~2 weeks stale — none of the recent changes (tool-UI docs, MCP streamable-http transport, MCP 2026-07-28 support matrices from #632/#633/#634) are deployed.

## Fix

Add the missing sidebar entry next to its siblings (in-memory / file / device / transforming) in the Swift storage section of `docs/astro.config.mjs`. Verified locally that the build completes after the fix.

--- a/docs/astro.config.mjs
+++ b/docs/astro.config.mjs
@@ -250,6 +250,7 @@ export default defineConfig({
 											{ label: 'File', slug: 'swift/storage/built-in/file' },
 											{ label: 'Device (SwiftData)', slug: 'swift/storage/built-in/device' },
 											{ label: 'Transforming (PII scrub)', slug: 'swift/storage/built-in/transforming' },
+											{ label: 'Summarizing', slug: 'swift/storage/built-in/summarizing' },
 										],
 									},
 									{ label: 'Custom store', slug: 'swift/storage/custom' },

